packages/patternfly-4/react-core/src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { DetailedHTMLProps, HTMLAttributes, ReactNode } from 'react';
+
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/patternfly/components/Badge/badge.css';
+
+const defaultProps = {
+  isRead: false,
+  children: '',
+  className: ''
+};
+
+export interface BadgeProps extends DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement> {
+  /**  Adds styling to the badge to indicate it has been read */
+  isRead: boolean;
+  /** content rendered inside the Badge */
+  children: ReactNode;
+  /** additional classes added to the Badge */
+  className?: string;
+}
+
+const Badge: React.FunctionComponent<BadgeProps> = (props) => (
```

Comment:
Should destructure and then use props directly
```suggestion
const Badge: React.FunctionComponent<BadgeProps> = ({ isRead, className, children, ...props }) => (
```

Reply:
@redallen This should work for typescript ({isRead, className, children, ...props}: BadgeProps).  I think the avatar one should be updated to this as well.

Reply:
Done for both.